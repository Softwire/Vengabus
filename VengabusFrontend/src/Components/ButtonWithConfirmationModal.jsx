import React from 'react';
import { Modal, Alert, Button, Tooltip, OverlayTrigger } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithConfirmationModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleOpen = () => {
        const afterOpen = (this.props.afterShowModalAction || (() => { }));
        this.setState({ show: true }, afterOpen);
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        const afterClose = (this.props.afterCloseModalAction || (() => { }));
        this.setState(defaultState, afterClose);
    }

    render() {
        const style = this.props.buttonDisabled ? {
            pointerEvents: 'none'
        } : {};

        const tooltip = (
            <Tooltip id="tooltip">
                <strong>{this.props.tooltipMessage}</strong>
            </Tooltip>
        );

        const button =
            <Button
                onClick={this.handleOpen}
                bsStyle={this.props.buttonStyle || "danger"}
                disabled={!!this.props.buttonDisabled}
                style={style}
            >
                {this.props.buttonText}
            </Button>;

        const buttonWithToolTip =
            <OverlayTrigger rootClose overlay={tooltip} placement="top">
                <div style={{ display: 'inline-block', cursor: 'not-allowed', float: 'right' }}>
                    {button}
                </div>
            </OverlayTrigger>;


        return (
            <React.Fragment>
                {this.props.tooltipMessage ? buttonWithToolTip : button}
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        {this.props.modalBody ?
                            <Alert bsStyle={this.props.buttonStyle ? this.props.buttonStyle : "danger"}>
                                {this.props.modalBody}
                            </Alert > : null
                        }
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText || "Cancel"}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment >
        );
    }
}

export { ButtonWithConfirmationModal };
