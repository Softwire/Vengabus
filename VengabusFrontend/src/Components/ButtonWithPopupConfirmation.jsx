import React from 'react';
import { Modal, Alert, Button } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithPopupConfirmation extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        this.props.closeModalAction();
        this.setState(defaultState);
    }

    handleOpening = () => {
        this.setState({ show: true });
        this.props.showModalAction();
    }

    render() {
        return (
            <React.Fragment>
                <Button onClick={this.handleOpening} bsStyle={this.props.buttonStyle}>
                    {this.props.buttonText}
                </Button>
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <Alert bsStyle="danger">
                            {this.props.modalBody}
                        </Alert >
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment>
        );
    }
}

export { ButtonWithPopupConfirmation };
