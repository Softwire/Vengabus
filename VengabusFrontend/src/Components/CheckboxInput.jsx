import React, { Component } from 'react';
import { css } from 'emotion';
import { Checkbox, Glyphicon, OverlayTrigger } from 'react-bootstrap';

/**
 * @prop {bool} data The data that is set by the checkbox.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the checkbox is changed.
 * @prop {string} text The name of the property that is being set by the checkbox.
 */
export class CheckboxInput extends Component {

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;

        return (
            <div>
                {
                    this.props.tooltip ? (
                        <p className={leftAlign}>
                            {this.props.text}
                            <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                <Glyphicon glyph="info-sign" className={glyphStyle} />
                            </OverlayTrigger>
                        </p>
                    ) : (
                            <p className={leftAlign}>{this.props.text}</p>
                        )
                }
                <Checkbox
                    className={leftAlign}
                    checked={this.props.data}
                    onChange={(event) => {
                        this.props.onChange(event.target.checked);
                    }}
                />
            </div>
        );
    }

}


