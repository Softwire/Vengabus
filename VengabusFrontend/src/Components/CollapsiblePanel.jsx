import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';

export class CollapsiblePanel extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            glyphicon: "plus"
        };
    }


    toggleGlyphicon = () => {
        this.setState({
            glyphicon: this.state.glyphicon === "plus" ? "minus" : "plus"
        });
    }

    render() {
        const panelStyle = css`
            .panel {
                margin-bottom: 10px;
            }
            .panel-heading {
                padding: 0 15px;
                color: #333;
                background-color: #f5f5f5;
                border-color: #ddd;
            }
            .panel-body { 
                padding: 0;
            }
            pre {
                background-color: #ffffff;
                border: none;
                margin: 0;
            }
            pre p {
                margin: 0;
            }
            
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const glyphiconStyle = css`
            .glyphicon-plus {
                margin-left: 1px;
                margin-right: 10px;
            }
            .glyphicon-minus {
                margin-right: 11px;
            }
        `;
        return (
            <Panel defaultExpanded={this.props.isDefaultExpanded} className={panelStyle} onToggle={() => this.toggleGlyphicon()} >
                <Panel.Toggle>
                    <Panel.Heading>
                        <Panel.Title className={glyphiconStyle} >
                            <Glyphicon glyph={this.state.glyphicon} /><h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        </Panel.Title>
                    </Panel.Heading>
                </Panel.Toggle>
                <Panel.Collapse>
                    <Panel.Body>{this.props.panelContent}</Panel.Body>
                </Panel.Collapse>
            </Panel >

        );
    }

}