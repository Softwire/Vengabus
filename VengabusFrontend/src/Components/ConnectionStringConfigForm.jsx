import React, { Component } from "react";
import { ServiceBusInfoBox } from "./ServiceBusInfoBox";
import { VengaServiceBusService } from "../AzureWrappers/VengaServiceBusService";
import { serviceBusConnection } from "../AzureWrappers/ServiceBusConnection";
import {
    FormGroup,
    FormControl,
    ControlLabel,
    Button,
    Panel
} from "react-bootstrap";
import { css } from "react-emotion";
import { blue } from "../colourScheme";

export const LOCAL_STORAGE_STRINGS = Object.freeze({
    ConnectionString: "connectionString",
    APIroot: "apiRoot"
});

/**
 *    Local Storage is accessible as a global window when run in the browser.
 *    This variable exists mostly to be a holder for this comment! The global
 *    variable will be stubbed in by the jest-localstorage-mock package, during testing.
 */
const localStorageAccessor = localStorage;

export class ConnectionStringConfigForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            connStringVal: "",
            APIroot: "",
            info: ""
        };
    }

    componentWillMount() {
        const localStorageConnectionString =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.ConnectionString) ||
            "";
        this.updateConString(localStorageConnectionString);
        const localStorageApiRoot =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.APIroot) ||
            "";
        this.updateAPIroot(localStorageApiRoot);
    }

    /** Updates the value of the connection string in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newConString The updated value of the connection string.  
     */
    updateConString = newConString => {
        this.setState({ connStringVal: newConString });
        serviceBusConnection.setConnectionString(newConString);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.ConnectionString,
            newConString
        );
    };

    /** Updates the value of the API root location in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newURI The updated value of the API root.  
     */
    updateAPIroot = newURI => {
        Math.random();
        this.setState({ APIroot: newURI });
        serviceBusConnection.setAPIroot(newURI);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.APIroot,
            newURI
        );
    }

    // Called whenever the value of the connection string input box changes.
    handleConnectionChange = event => {
        this.updateConString(event.target.value);
    };

    // Called whenever the value of the API server input box changes.
    handleAPIChange = event => {
        this.updateAPIroot(event.target.value);
    }

    /**
     * Updates the info in the sidebar based on the current status of VengaServiceBusService.
     * Called whenver the connect button is pressed.
     */
    submitConnectionStringClick = () => {
        const infoPromise = VengaServiceBusService.getServiceBusProperties(
            this.state.connStringVal
        );
        infoPromise
            .then(response => {
                this.setState({
                    info: response
                });
            })
            .catch(error => {
                console.log(error);
            });
    };

    render() {
        const inputStyle = css`
      color: black;
    `;

        const buttonStyle = css`
      color: black;
      margin: 5px;
    `;

        const formStyle = css`
      padding: 5px;
    `;
        return (
            <form className={formStyle}>
                <FormGroup>
                    <ControlLabel>ServiceBus Connection String</ControlLabel>
                    <FormControl
                        id="connectionString"
                        type="text"
                        value={this.state.connStringVal}
                        placeholder="Enter Connection String"
                        onChange={this.handleConnectionChange}
                    />
                    <FormControl.Feedback />

                    <ControlLabel>ServiceBus API Server String</ControlLabel>
                    <FormControl
                        id="APILocationForm"
                        type="text"
                        value={this.state.APIroot}
                        placeholder="Enter API Server Location"
                        onChange={this.handleAPIChange}
                    />
                    <FormControl.Feedback />

                </FormGroup>
                <Button
                    className={buttonStyle}
                    onClick={this.submitConnectionStringClick}
                >
                    Connect
                </Button>
                {
                    //buttons want to grip on to the top of things not pretty so add a break to separate
                }
                <div>
                    <br />
                </div>
                <ServiceBusInfoBox
                    connStringVal={this.state.connStringVal}
                    info={this.state.info}
                />
            </form>
        );
    }
}
