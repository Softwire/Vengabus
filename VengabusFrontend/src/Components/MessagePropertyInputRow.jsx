import React, { Component } from 'react';
import { css } from 'react-emotion';
import {
    FormGroup,
    FormControl,
    Button,
    DropdownButton,
    MenuItem
} from "react-bootstrap";
import Select from 'react-select';

/** Required props:
 * @prop {string} propertyName Name of the property.
 * @prop {string} propertyValue Value of the property.
 * @prop {int} index To be able to identify its place.
 * @prop {function} getValidNameState Function that validates this input.
 * @prop {function} handlePropertyNameChange Function that is called when propertyName is edited.
 * @prop {function} handlePropertyValueChange Function that is called when propertyValue is edited.
 * @prop {function} deleteRow Function that is called when the delete button is pressed.
 * @prop {string[]} permittedValues A list of property values that are allowed, if not present, any values will be allowed. 
 */
export class MessagePropertyInputRow extends Component {
    constructor(props) {
        super(props);
    }

    FieldGroup({ id, validation, ...props }) {
        return (
            <FormGroup controlId={id} validationState={validation}>
                <FormControl {...props} />
            </FormGroup>
        );
    }

    render() {
        const formStyle = css`
            padding: 5px;
            width: 47%;
            float: left;
        `;
        const deleteButtonStyle = css`
            padding: 5px;
            width: 5%; 
            float: left;
        `;
        const propertyName = this.props.propertyName;
        const index = this.props.index;
        const permittedValues = this.props.permittedValues;
        let permittedValueMenuItems;
        if (permittedValues) {
            permittedValueMenuItems = permittedValues.map((permittedValue) => { return { value: permittedValue, label: permittedValue }; });
        }

        return (
            <div>
                <form className={formStyle}>
                    {this.props.permittedValues ? (
                        <Select
                            title="Choose a property"
                            key={index}
                            id={`property-dropdown-${index}`}
                            options={permittedValueMenuItems}
                            value={propertyName ? { value: propertyName, label: propertyName } : undefined}
                            onChange={(event) => this.props.handlePropertyNameChange(event.value, index)}
                        />

                    ) : (
                            <this.FieldGroup
                                id="formControlsText"
                                validation={this.props.getValidNameState(index)}
                                key={index}
                                type="text"
                                placeholder="Enter property name"
                                value={propertyName}
                                onChange={(event) => this.props.handlePropertyNameChange(event.target.value, index)}
                            />
                        )
                    }
                </form>
                <form className={formStyle}>
                    <this.FieldGroup
                        id="formControlsText"
                        key={index}
                        type="text"
                        placeholder="Enter property value"
                        value={this.props.propertyValue}
                        onChange={(event) => this.props.handlePropertyValueChange(event, index)}
                    />
                </form >
                <div className={deleteButtonStyle}>
                    <Button className="delete-button" bsStyle="danger" onClick={() => this.props.deleteRow(index)}>Delete</Button>
                </div>
            </div >
        );
    }

}