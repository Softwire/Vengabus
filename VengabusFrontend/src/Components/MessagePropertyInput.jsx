import React, { Component } from 'react';
import { css } from 'react-emotion';
import { Button } from "react-bootstrap";
import { MessagePropertyInputRow } from './MessagePropertyInputRow';


export class MessagePropertyInput extends Component {
    constructor(props) {
        super(props);

        this.state = {
            propertyNames: [],
            propertyValues: []
        };
    }

    handlePropertyNameChange = (event, position) => {
        let newPropertyNames = this.state.propertyNames;
        newPropertyNames[position] = event.target.value;
        this.setState({ propertyNames: newPropertyNames });
    };

    handlePropertyValueChange = (event, position) => {
        let newPropertyValues = this.state.propertyValues;
        newPropertyValues[position] = event.target.value;
        this.setState({ propertyValues: newPropertyValues });
    };

    addNewProperty = () => {
        let newPropertyValues = this.state.propertyValues;
        newPropertyValues[newPropertyValues.length] = "";
        let newPropertyNames = this.state.propertyNames;
        newPropertyNames[newPropertyNames.length] = "";
        this.setState({
            propertyValues: newPropertyValues,
            propertyNames: newPropertyNames
        });
    }

    submit = () => {
        let properties = {};
        for (let i = 0; i < this.state.propertyNames.length; i++) {
            const propertyNames = this.state.propertyNames;
            const propertyValues = this.state.propertyValues;
            if (propertyNames[i] && propertyValues[i] && !properties.hasOwnProperty(propertyNames[i])) {
                //Prevent the user from inputting reserved property names
                if (propertyNames[i] !== 'body' && propertyNames[i] !== 'queuename' && propertyNames[i] !== 'SAS') {
                    properties[propertyNames[i]] = propertyValues[i];
                }
            }
        }
        this.props.submitMessage(properties);
    }



    getValidNameState = (i) => {
        let name = this.state.propertyNames[i];
        if (name === 'body' || name === 'queuename' || name === 'SAS' || name.length === 0) {
            return 'error';
        } else {
            return null;
        }
    }

    deleteRow = (i) => {
        const newPropertyNames = this.state.propertyNames.slice();
        newPropertyNames.splice(i, 1);
        const newPropertyValues = this.state.propertyValues.slice();
        newPropertyValues.splice(i, 1);
        this.setState({
            propertyNames: newPropertyNames,
            propertyValues: newPropertyValues
        });
    }

    render() {
        let inputs = [];
        for (let i = 0; i < this.state.propertyNames.length; i++) {
            inputs.push(
                <MessagePropertyInputRow
                    propertyName={this.state.propertyNames[i]}
                    propertyValue={this.state.propertyValues[i]}
                    index={i}
                    getValidNameState={this.getValidNameState}
                    handlePropertyNameChange={this.handlePropertyNameChange}
                    handlePropertyValueChange={this.handlePropertyValueChange}
                    deleteRow={this.deleteRow}
                />
            );
        }

        return (
            <div>
                {inputs}
                <form>
                    <Button
                        id="addNewPropertyButton"
                        onClick={this.addNewProperty}
                    >
                        Add New Property
                     </Button>
                </form>
                <form>
                    <Button
                        id="submitButton"
                        onClick={this.submit}
                    >
                        Submit
                     </Button>
                </form>
            </div>
        );
    }

}
