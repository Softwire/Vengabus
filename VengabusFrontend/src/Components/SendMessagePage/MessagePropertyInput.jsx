import React, { Component } from 'react';
import { MessagePropertyInputRow } from './MessagePropertyInputRow';
import _ from 'lodash';

/** Renders a set of inputs (either dropdown or normal)
 * @prop {string[]} properties Contains all of the names and values of properties.
 * @prop {function} handlePropertyNameChange Called when a value in the left column is changed.
 * @prop {function} handlePropertyValueChange Called when a value in the right column is changed.
 * @prop {function} deleteRow Called when a delete button is pressed.
 * @prop {string[]} permittedValues Contains the values for the dropdown list. If unspecified, then any user input is accepted in the left column.
 */
export class MessagePropertyInput extends Component {

    /**
     * Checks whether the name of a user defined property is valid, i.e. neither empty nor a duplicate.
     * @param {integer} index The index of the name to check.
     * @return {string} 'error' if the name is invalid, or null otherwise.
     */
    isPropertyNameValid = (index) => {
        let name = this.props.properties[index].name;
        let occurencesInUserDefinedProperties = _(this.props.properties)
            .filter((current) => current.name === name)
            .size();
        let occurencesInPredefinedProperties = _(this.props.reservedPropertyNames)
            .filter((current) => current === name)
            .size();
        if (!name) {
            return false;
        }
        let isPropertyValid = true;
        if (occurencesInUserDefinedProperties > 1) {
            let newWarning = "Warning: repetitive property name: '" + name + "'";
            if (this.warnings.indexOf(newWarning) === -1) {
                this.warnings.push(newWarning);
            }
            isPropertyValid = false;
        }
        if (occurencesInPredefinedProperties > 0) {
            let newWarning = "Warning: custom property '" + name + "' is potentially a predefined property";
            if (this.warnings.indexOf(newWarning) === -1) {
                this.warnings.push(newWarning);
            }
            isPropertyValid = false;
        }
        return isPropertyValid;
    }

    render() {
        let inputs = [];
        this.warnings = [];
        let remainingPermittedValues;
        if (this.props.permittedValues) {
            remainingPermittedValues = [...this.props.permittedValues];
        }
        for (let i = 0; i < this.props.properties.length; i++) {
            //Remove properties that have already been selected from the dropdown
            const indexOfExistingValue = remainingPermittedValues ? remainingPermittedValues.indexOf(this.props.properties[i].name) : -1;
            if (indexOfExistingValue !== -1) {
                remainingPermittedValues.splice(indexOfExistingValue, 1);
            }
        }
        for (let i = 0; i < this.props.properties.length; i++) {
            inputs.push(
                <MessagePropertyInputRow
                    propertyName={this.props.properties[i].name}
                    propertyValue={this.props.properties[i].value}
                    index={i}
                    key={i}
                    isPropertyNameValid={(index) => this.isPropertyNameValid(index) ? null : 'error'}
                    handlePropertyNameChange={this.props.handlePropertyNameChange}
                    handlePropertyValueChange={this.props.handlePropertyValueChange}
                    deleteRow={this.props.deleteRow}
                    permittedValues={remainingPermittedValues}
                />
            );
        }

        if (this.props.reportWarnings) {
            this.props.reportWarnings(this.warnings);
        }

        return <div>{inputs}</div>;
    }

}
