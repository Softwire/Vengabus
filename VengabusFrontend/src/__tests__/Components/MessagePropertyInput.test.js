import 'jest-localstorage-mock';
import { shallow, mount, configure } from 'enzyme';
import Adaptor from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import React from 'react';
import { Button } from "react-bootstrap";
import { MessagePropertyInput } from '../../Components/MessagePropertyInput';
import { MessagePropertyInputRow } from '../../Components/MessagePropertyInputRow';

configure({ adapter: new Adaptor() });

it('Correctly creates the properties of a message', () => {
    let spy = jest.fn();
    let wrapper = shallow(
        <MessagePropertyInput
            submitMessage={spy}
        />);
    wrapper.setState({
        propertyNames: ["test1", "test2"],
        propertyValues: ["any value 1", "any value 2"]
    });
    let submitButton = wrapper.find("#submitButton");
    submitButton.simulate('click');
    expect(spy).toBeCalledWith({
        test1: "any value 1",
        test2: "any value 2"
    });
});

it('Correctly filters reserved property names', () => {
    let wrapper = shallow(
        <MessagePropertyInput
            submitMessage={(message) => {
                expect(message.body).toBeUndefined();
            }}
        />);
    wrapper.setState({
        propertyNames: ["test1", "body"],
        propertyValues: ["t1", "any value"]
    });
    let submitButton = wrapper.find("#submitButton");
    submitButton.simulate('click');
});

it('Correctly filters empty property names', () => {
    let wrapper = shallow(
        <MessagePropertyInput
            submitMessage={(message) => {
                expect(message[""]).toBeUndefined();
            }}
        />);
    wrapper.setState({
        propertyNames: ["test1", ""],
        propertyValues: ["t1", "any value"]
    });
    let submitButton = wrapper.find("#submitButton");
    submitButton.simulate('click');
});

it('Correctly filters duplicate property names', () => {
    let wrapper = shallow(
        <MessagePropertyInput
            submitMessage={(message) => {
                expect(Object.keys(message).length).toEqual(2);
            }}
        />);
    wrapper.setState({
        propertyNames: ["test1", "test2", "test2"],
        propertyValues: ["any value", "any value2", "any value 3"]
    });
    let submitButton = wrapper.find("#submitButton");
    submitButton.simulate('click');
});

it('Deletes a property when the delete button is pressed', () => {
    let wrapper = mount(<MessagePropertyInput />);
    wrapper.setState({
        propertyNames: ["test1", "test2"],
        propertyValues: ["any value", "any value2"]
    });
    console.log(wrapper.find(MessagePropertyInputRow).at(0));
    let deleteButton = wrapper.find(MessagePropertyInputRow).at(0).find(Button);
    deleteButton.simulate('click');
    expect(wrapper.find(MessagePropertyInputRow).length).toEqual(1);
});

it('renders correctly', () => {
    let messagePropertyInput = renderer.create(
        <MessagePropertyInput
            submitMessage={jest.fn()}
        />);
    expect(messagePropertyInput.toJSON()).toMatchSnapshot();
});