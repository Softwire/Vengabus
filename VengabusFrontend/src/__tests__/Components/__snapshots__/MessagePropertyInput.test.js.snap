// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form>
    <button
      className="btn btn-default"
      disabled={false}
      id="addNewPropertyButton"
      onClick={[Function]}
      type="button"
    >
      Add New Property
    </button>
  </form>
  <form>
    <button
      className="btn btn-default"
      disabled={false}
      id="submitButton"
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </form>
</div>
`;
